subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformFilteringIntegrationTest.groovy

```diff
@@ -132,4 +134,114 @@ class JUnitPlatformFilteringIntegrationTest extends JUnitPlatformIntegrationSpec
             .assertTestPassed('superTest')
     }
 
+    def 'does not exclude tests with a non-standard test source if filter matches nothing'() {
+        given:
+        buildFile << """
+            dependencies {
+                testImplementation 'com.tngtech.archunit:archunit-junit5:${LATEST_ARCHUNIT_VERSION}'
+            }
+
+            test {
+                filter {
+                    excludeTestsMatching "*notMatchingAnythingSoEverythingShouldBeRun"
+                }
+            }
+        """
+        file('src/test/java/DeclaresTestsAsFieldsNotMethodsTest.java') << '''
+            import com.tngtech.archunit.junit.AnalyzeClasses;
+            import com.tngtech.archunit.junit.ArchTest;
+            import com.tngtech.archunit.lang.ArchRule;
+
+            import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
+
+            @AnalyzeClasses(packages = "example")
+            class DeclaresTestsAsFieldsNotMethodsTest {
+                // this will create a JUnit Platform TestDescriptor with neither a Class- nor a MethodSource
+                @ArchTest
+                static final ArchRule example = classes().should().bePublic();
+            }
+        '''
+
+        when:
+        succeeds('test')
+
+        then:
+        new DefaultTestExecutionResult(testDirectory)
+            .assertTestClassesExecuted('DeclaresTestsAsFieldsNotMethodsTest')
+            .testClass('DeclaresTestsAsFieldsNotMethodsTest')
+            .assertTestCount(1, 0, 0)
+            .assertTestPassed('example')
+    }
+
+    def 'does not exclude tests with a non-standard test source if filter matches field name'() {
```

Comment:
In case we later go back to this, I think we should document here that this test documents the status quo but it would be desirable to change the behavior if we add support for handling fields.

Reply:
I'll add a note before I merge this.  Thanks for the reviews!